src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -29,6 +31,81 @@ constexpr auto DELAYED_EVENTS_QUEUE_PATH = "queue/vd/delayed";
 constexpr auto SLEEP_RETRY_THREADS_SEC = 60;
 
 int SOCKET_WAIT = 0;
+constexpr auto COMPRESSED_DB_PATH {"vd_1.0.0_vd_4.8.0.tar.xz"};
+constexpr auto DECOMPRESSED_DB_PATH {"vd_1.0.0_vd_4.8.0.tar"};
+constexpr auto VD_STATE_QUEUE_PATH = "queue/vd/state_track";
+constexpr auto VD_KEYSTORE_PATH = "queue/keystore";
+constexpr auto VD_DATABASE_PATH {"queue/vd"};
+constexpr auto VD_DATABASE_VERSION_KEY {"installed_content"};
+
+bool VulnerabilityScannerFacade::decompressDatabase(const std::string& databaseVersion)
+{
+    bool ret = false;
+
+    // Check database version and presence. It will attempt to decompress the database
+    // if the version doesn't not match, the state_track does not have information
+    // or there's no VD DB.
+    if ((databaseVersion.compare(__ossec_version) != 0 || databaseVersion.empty() ||
+         !std::filesystem::exists(VD_DATABASE_PATH)) &&
+        std::filesystem::exists(TMP_DIR))
+    {
+        // Search in path for a compressed database file.
+        std::regex regexPattern(VD_COMPRESSED_FILE_PATTERN);
+        std::string compressedFilePath;
+        size_t compressedFilesNumber = 0;
+        for (const auto& item : std::filesystem::directory_iterator(TMP_DIR))
```

Comment:
Use COMPRESSED_DB_PATH, What would happen if you upgrade without starting Wazuh and you have several files in the tmp folder? How do we guarantee that the order is correct?

Reply:
Reverted the regex logic